### PR TITLE
Document exit codes for podman exec

### PR DESCRIPTION
**Pull request by rhatdan**
Also fix podman run exit codes to show real messages when failures
happen.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @mheon @baude @giuseppe PTAL

File: cmd/podman/run.go [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package main
 
 import (
+	"fmt"
```
Comment (TomSweeneyRedHat) [inline cmd/podman/run.go]: leftover from testing?

File: docs/podman.1.md [hunk 2]
```diff
@@ -103,22 +103,23 @@ the exit codes follow the `chroot` standard, see below:
...
     $ podman run busybox foo; echo $?
-    # exec: "foo": executable file not found in $PATH
-      podman: Error response from daemon: Contained command not found or does not exist
-      127
+    Error: container create failed: time="2019-06-12T05:01:25-04:00" level=error msg="container_linux.go:346: starting container process caused \"exec: \\\"foo\\\": executable file not found in $PATH\"\n"
+    container_linux.go:346: starting container process caused "exec: \"foo\": executable file not found in $PATH"
+: internal libpod error
+    127
 
 **_Exit code_** of **_container command_** otherwise
```
Comment (TomSweeneyRedHat) [inline docs/podman.1.md]: Probably should change this to "contained command" too

File: libpod/oci_linux.go [hunk 3]
```diff
@@ -418,7 +418,7 @@ func (r *OCIRuntime) createOCIContainer(ctr *Container, cgroupParent string, res
 		logrus.Debugf("Received container pid: %d", ss.si.Pid)
 		if ss.si.Pid == -1 {
 			if ss.si.Message != "" {
-				return errors.Wrapf(ErrInternal, "container create failed: %s", ss.si.Message)
+				return errors.Wrapf(ErrInternal, ss.si.Message)
```
Comment (TomSweeneyRedHat) [inline libpod/oci_linux.go]: Personally I wouldn't have made this change.

Comment (rhatdan) [reply libpod/oci_linux.go]: This is in hold right now, while waiting for @giuseppe Patch to get merged, I actually want this to go further. 
This is not an Internal Err, and it is not failing to create the container.  The container was created successfully and the command failed, within the container.  So we are reporting two incorrect messages up the stack.

**Comment by mheon** [comment]
If you can rebase and hit comments I'd love this in 1.4.1

File: cmd/podman/run.go (hunk 1 above)
Comment (rhatdan) [reply cmd/podman/run.go]: Fixed

**Comment by rhatdan** [comment]
@mheon Rebased.

File: docs/podman.1.md [hunk 4]
```diff
@@ -103,24 +103,21 @@ the exit codes follow the `chroot` standard, see below:
 **_125_** if the error is with podman **_itself_**
 
     $ podman run --foo busybox; echo $?
-    # flag provided but not defined: --foo
-      See 'podman run --help'.
+    Error: unknown flag: --foo
       125
 
 **_126_** if executing a **_container command_** and the the **_command_** cannot be invoked
```
Comment (TomSweeneyRedHat) [inline docs/podman.1.md]: crud, didn't spot this earlier.  _container command_ here, _contained command_ below.  Can we make them all the same?  I like the contained thought fwiw.

File: docs/podman.1.md [hunk 5]
```diff
@@ -103,24 +103,21 @@ the exit codes follow the `chroot` standard, see below:
...
 
     $ podman run busybox /etc; echo $?
-    # exec: "/etc": permission denied
-      podman: Error response from daemon: Contained command could not be invoked
+    Error: container_linux.go:346: starting container process caused "exec: \"/etc\": permission denied": OCI runtime error
       126
 
 **_127_** if executing a **_container command_** and the the **_command_** cannot be found
```
Comment (TomSweeneyRedHat) [inline docs/podman.1.md]: my eyeballs missed this one earlier too.

**Comment by TomSweeneyRedHat** [comment]
Two nits.  Otherwise LGTM.  If you want to push this through and circle back for the nits, fine by me.

**Comment by rhatdan** [comment]
Lets merge and then You or I could open a PR to fix them.  I just don't want to hold @mheon up on cutting a new release waiting for this.

**Comment by mheon** [comment]
LGTM, tests are green, merging
/lgtm

**Pull request merged**